workers/unity/Packages/com.improbable.gdk.core/Tests/Editmode/Systems/SpatialOSReceiveSystemTests.cs

```diff
@@ -0,0 +1,738 @@
+            receiveSystem.AddDispatcherHandler(secondComponentDispatcher);
+
+            var requestTracker = world.GetOrCreateManager<CommandRequestTrackerSystem>();
+            createEntityStorage = requestTracker.GetCommandStorageForType<WorldCommands.CreateEntity.Storage>();
+            deleteEntityStorage = requestTracker.GetCommandStorageForType<WorldCommands.DeleteEntity.Storage>();
+            reserveEntityIdsStorage = requestTracker.GetCommandStorageForType<WorldCommands.ReserveEntityIds.Storage>();
+            entityQueryStorage = requestTracker.GetCommandStorageForType<WorldCommands.EntityQuery.Storage>();
+        }
+
+        [OneTimeTearDown]
+        public void OneTimeTearDown()
+        {
+            world.Dispose();
+        }
+
+        [TearDown]
+        public void TearDown()
+        {
+            foreach (var entity in worker.EntityIdToEntity.Values)
+            {
+                try
+                {
+                    WorldCommands.DeallocateWorldCommandRequesters(entityManager, entity);
+                }
+                catch (Exception e)
+                {
+                }
+            }
+
+            worker.EntityIdToEntity.Clear();
+
+            using (var allEntities = entityManager.GetAllEntities())
+            {
+                foreach (var entity in allEntities.Where(x => x != worker.WorkerEntity))
+                {
+                    entityManager.DestroyEntity(entity);
+                }
+            }
+
+            firstComponentDispatcher.Reset();
+            logDispatcher.ClearExpectedLogs();
+
+            createEntityStorage.CommandRequestsInFlight.Clear();
+            deleteEntityStorage.CommandRequestsInFlight.Clear();
+            reserveEntityIdsStorage.CommandRequestsInFlight.Clear();
+            entityQueryStorage.CommandRequestsInFlight.Clear();
+
+            WorldCommands.CreateEntity.ResponsesProvider.CleanDataInWorld(world);
+            WorldCommands.EntityQuery.ResponsesProvider.CleanDataInWorld(world);
+            WorldCommands.DeleteEntity.ResponsesProvider.CleanDataInWorld(world);
+            WorldCommands.ReserveEntityIds.ResponsesProvider.CleanDataInWorld(world);
+        }
+
+        [Test]
+        public void OnAddEntity_should_add_entity_and_command_components()
+        {
+            using (var wrappedOp = WorkerOpFactory.CreateAddEntityOp(TestEntityId))
+            {
+                receiveSystem.OnAddEntity(wrappedOp.Op);
+            }
+
+            Assert.IsTrue(worker.TryGetEntity(new EntityId(TestEntityId), out var entity));
+
+            var id = new SpatialEntityId(); // Default value
+            Assert.DoesNotThrow(() => { id = entityManager.GetComponentData<SpatialEntityId>(entity); });
+            Assert.AreEqual(TestEntityId, id.EntityId.Id);
+
+            ComponentType[] worldCommandComponentTypes =
+            {
+                ComponentType.Create<WorldCommands.CreateEntity.CommandSender>(),
```

Comment:
do we need `Create`? or could it be `ReadOnly`

Reply:
Doesn't really matter tbh

Reply:
Create is less keystrokes 😛

Reply:
fair enough :P